### DISCUSSION TITLE
Recovering workspace after crash

### DISCUSSION THREAD

**Discussion by RobinM92** [Open-Systems-Pharmacology/Forum#1303] [Questions & Problems] [answered]
Dear community, 
When working with larger parameter identifications/much observed data, it happens that PKSim or MoBi freezes up and crashes. This can be especially frustrating when the project was not saved over some time (e.g. due to painstakingly creating many simulations to put together in a PI step) as there does not seem to be an autosave nor recovery function. Is there any way to recover lost work when things like this happen?

All the best,
Robin

**Comment by Yuri05** [comment, marked as answer]
Unfortunately there is no possibility to recover unsaved project if application has crashed.
For now I can only suggest that you save your project before starting a PI.

Is it possible to attach one of the projects where the software crashes during the PI?
What kind of error do you get (out of memory, other)?

**Reply by msevestre** [reply]
Maybe we should trigger a save before starting PI or SA? Or a message suggesting saving at least and an option to autosave when running PI or SA

**Reply by RobinM92** [reply]
Thanks Yuri!
Actually, the software already crashes before the PI itself, while inputting the simulations and observed data to be used by the tool. This has happened both in PKSim and Mobi when relatively large dataset/simulations are used, so I suspect it is a virtual memory crash. No error message pops up, the software just closes suddenly (sometimes after being frozen for a short while).

**Reply by msevestre** [reply]
@RobinM92 Do you have an example of such a project
And can you give us some details on your config (RAM in particular). Are you using a VM ?

**Reply by wilbertdew** [reply]
I have had similar issues in the past, but I have no recent examples. One thing that seemed to help is to avoid saving on a network drive. At least this reduced saving time of large files sometimes from 20 minutes to half a minute, and I think it also improved stability.

**Reply by Yuri05** [reply]
@RobinM92 Can you check you windows event log?
* Start the event viewer (e.g. just type "Event in Windows search - it will be shown as one of the first items)
* Go to Windows Logs/Applications
* Click on "Filter Current Log"
* Select event level "Critical" and "Error"
* Open the list for the "Event Sources" and check (if you find them in the list)
  * .NET runtime
  * PK-Sim
  * MoBi
![grafik](https://user-images.githubusercontent.com/25061876/191322187-6f677c37-9b3b-445c-8e60-d389903b2769.png)

After this, go through the filtered events and look for those related to PK-Sim and MoBi. Would look like this:
![grafik](https://user-images.githubusercontent.com/25061876/191323646-79a411ee-871e-44be-929a-79dc29bb1bb9.png)

Can you paste the content of the highlighted panel for some of the recent crashes?
Or (better): export the filtered event log (Action=>Save selected items) in the event files format (***.evtx**), zip it and attach here.

**Reply by Yuri05** [reply]
@RobinM92 Also please try to increase Windows limits for GDI/User objects:
 * start `Regedit` as administrator
 * navigate to **HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Windows**
 * increase value of **GDIProcessHandleQuota** (e.g. double it; max. value is _65536_ (**decimal**))
 * increase value of **USERProcessHandleQuota** (e.g. double it; max. value is _65536_ (**decimal**))
 * **restart the computer** (!)
 
![grafik](https://user-images.githubusercontent.com/25061876/191325596-0d32e3c0-5012-4a51-8be7-acd669333505.png)

**Reply by RobinM92** [reply]
Thanks for all the suggestions!
I am not able to share the current project, but my current machine runs with an i7 processor at 32GB RAM, so I would be surprised if that would not be enough. I work locally and commit afterwards using a version control software, so I also do not think that network limitations play a role here. It does seem that the VMEM usage increases when I use PKSim/MoBi, so perhaps that reaches some kind of limit. 

I have attached two recent crash logs from the windows event log, both coming from .NET runtime. There are no recent logs for PKSim or MoBi.

[Events.zip](https://github.com/Open-Systems-Pharmacology/Forum/files/9632211/Events.zip)

**Reply by Yuri05** [reply]
Looking at your crash reports: I think you are indeed out of handles limit and increasing it as described above would help.
Once done, you can start PK-Sim or MoBi and open the task manager and watch the number of graphical (GDI) and User Objects.
If you see that they exceed the old limits - at this timepoint PK-Sim/MoBi would crash with the old settings.

![grafik](https://user-images.githubusercontent.com/25061876/191933069-0c93f900-6913-46c1-a7dd-b8437c05cc88.png)

**Reply by RobinM92** [reply]
So I increased (doubled) the number of GDI and User Objects, and the same crash happened again while just adapting some doses in a very long (39 dose events) simulation. Very annoying as everything just closes without any warning... I can only assumed that this is not the desired behaviour of the software?

**Reply by Yuri05** [reply]
Try setting the values to Max (s. above)

**Comment by PavelBal** [comment]
This must be the same problem as here:

https://github.com/Open-Systems-Pharmacology/MoBi/issues/469

https://github.com/Open-Systems-Pharmacology/PK-Sim/issues/1713

**Reply by RobinM92** [reply]
Seems like it!
